src/main/java/com/gmail/danadiadius/technicians/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,41 @@
+package com.gmail.danadiadius.technicians.web.filter;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterConfig;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletResponse;
+import javax.servlet.ServletRequest;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+public class AuthenticationFilter implements Filter {
+    private Set<String> allowedUrls = new HashSet<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        allowedUrls.add("/");
+        allowedUrls.add("/index");
+        allowedUrls.add("/login");
+        allowedUrls.add("/registration");
```

Comment:
Maybe it is a good idea to store these URLs in a file, just to not hardcode them.